homeassistant/components/stream/hls.py

```diff
@@ -0,0 +1,128 @@
+"""
+Provide functionality to stream HLS.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/hls
+"""
+import datetime
+
+from aiohttp import web
+
+from .core import StreamView, StreamOutput
+
+
+async def async_setup_hls(hass):
+    """Set up api endpoints."""
+    hass.http.register_view(HlsPlaylistView())
+    hass.http.register_view(HlsSegmentView())
+    return '{}/api/hls/{}/playlist.m3u8'
+
+
+class HlsPlaylistView(StreamView):
+    """Stream view to serve a M3U8 stream."""
+
+    url = r'/api/hls/{token:[a-f0-9]+}/playlist.m3u8'
+    name = 'api:stream:hls:playlist'
+    cors_allowed = True
+
+    async def handle(self, request, stream, sequence):
+        """Return m3u8 playlist."""
+        renderer = M3U8Renderer(stream)
+        track = stream.add_provider(HlsStreamOutput())
+        stream.start()
+        # Wait for a segment to be ready
+        if not track.segments:
+            await track.recv()
+        headers = {
+            'Content-Type': 'application/vnd.apple.mpegurl'
+        }
+        return web.Response(body=renderer.render(
+            track, datetime.datetime.utcnow()
+            ).encode("utf-8"), headers=headers)
+
+
+class HlsSegmentView(StreamView):
+    """Stream view to serve a MPEG2TS segment."""
+
+    url = r'/api/hls/{token:[a-f0-9]+}/segment/{sequence:\d+}.ts'
+    name = 'api:stream:hls:segment'
+    cors_allowed = True
+
+    async def handle(self, request, stream, sequence):
+        """Return mpegts segment."""
```

Comment:
Stale comment

Reply:
It applies.  This is returning the `.ts` segment (also known as `mpegts`).